db/python/layers/seqr.py

```diff
@@ -356,9 +367,50 @@ async def sync_individual_metadata(
             f'Uploaded individual metadata for {len(processed_records)} individuals'
         ]
 
+    def check_updated_sequencing_group_ids(self, sequencing_group_ids: set[int], es_index_analyses: list[AnalysisInternal]):
+        """Check if the sequencing group IDs have been updated"""
+        messages = []
+        if sequencing_group_ids:
+            sequencing_groups_in_new_index = set(
+                es_index_analyses[-1].sequencing_group_ids
```

Comment:
I would re-sort this at the top, it's a small operation and less dangerous than presuming it's sorted. (I know your usage is sorted on the way in, but there's no other condition that states this on this function)